Autostart ircbot based on environment variable

Olen opened the pull request:
This will ensure that the irc-bot is started if the enviroment-variable LIBRENMS_IRC_SERVICE is set to "1".

You still need to enable irc_alert in the config to use the ircbot for alerting.

Olen committed:
Autostart ircbot if irc_alert is set to true

This will ensure that the irc-bot is started if irc_alert is configured.

crazy-max commented on rootfs/etc/cont-init.d/04-svc-main.sh:
```diff
@@ -109,3 +109,14 @@ with-contenv
+mkdir -p /etc/services.d/ircbot
+cat >/etc/services.d/ircbot/run <<EOL
+#!/usr/bin/with-contenv sh
+if [ $(lnms config:get irc_alert) == "true" ]; then
```
Don't think we should check if irc alert is enabled in service run

crazy-max commented on rootfs/etc/cont-init.d/04-svc-main.sh:
```diff
@@ -109,3 +109,14 @@ with-contenv
+cat >/etc/services.d/ircbot/run <<EOL
+#!/usr/bin/with-contenv sh
+if [ $(lnms config:get irc_alert) == "true" ]; then
+  s6-setuidgid 1000:992
```
Any reason to use this specific UID/GID?

Olen replied:
Sorry, those ar the uids/gids I run with, but they are abviously not universal and should be taken from the environment.

crazy-max commented on rootfs/etc/cont-init.d/04-svc-main.sh:
```diff
@@ -109,3 +109,14 @@ with-contenv
+  s6-setuidgid 1000:992
+  /opt/librenms/irc.php
+fi
+sleep 60
```
?

Olen replied:
Added this because of auto-restart.  If something is wrong and the bot is unable to connect, it will retry continously, spamming the logs and using a lot of resources, and possibly get you banned from irc-servers.

crazy-max reviewed (changes requested):
Looking at this service I think we should have a new env var named smth like `LIBRENMS_IRC_SERVICE` instead of relying on `irc_alert` config.

Olen committed:
Updated based on review

- Changed startup to depend on nevironment variable
- Fixed uid/gid
- Shortened final sleep before restart (not sure about this one)

Olen commented:
Seems like the bot is unable to send alerts.  Possibly because the alerts are triggered by the dispatcher container, which does not have access to the .ircbot.alert file.
I will investigate a bit more.

Olen commented:
This PR needs to be merged to make alerting work in the container based environment.

https://github.com/librenms/librenms/pull/14647

crazy-max commented:
> This PR needs to be merged to make alerting work in the container based environment.
> 
> [librenms/librenms#14647](https://github.com/librenms/librenms/pull/14647)

Moving to draft while waiting for the upstream PR to be merged

murrant commented:
Approved upstream, will be in 23.5

Olen commented:
Now that 23.5 has been released.  This should be ready.
